### PR TITLE
Improve presentation of trace attributes

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

Improves the presentation of trace attributes by:

* Format `timestamp` as a date
* Move important attributes to the top: `timestamp`, `span_name`, `duration`, `service_name`, `environment`, `service_version`
* Remove `projectID`
* Remove empty values (`""`)
* Move `traceAttributes` properties to the top level and remove that key
* Demote default OTEL attributes: `host`, `os`, `process`
* Convert stringified numbers to numbers

**Before**

Screenshot 2024-01-18 at 11 39 32 PM

**After**

image

## How did you test this change?

Local + PR preview click testing.

## Are there any deployment considerations?

N/A - all client-side changes.

## Does this work require review from our design team?

N/A

### CODE REVIEW

File: frontend/src/pages/Traces/TraceSpanAttributes.tsx [hunk 1]
```diff
@@ -1,14 +1,74 @@
...
+	let attributes: { [key: string]: any } = { ...span }
 
 	// Drop any attributes we don't want to display
 	delete attributes.__typename
 	delete attributes.children
+	delete (attributes as any).projectID
+
+	// Convert timestamp to a formatted date/time
+	if (attributes.timestamp) {
+		attributes.timestamp = new Date(attributes.timestamp).toLocaleString()
```
Comment (Vadman97) [inline frontend/src/pages/Traces/TraceSpanAttributes.tsx:18]: could we still keep nanosecond precision? would be nice to preserve that for people trying to compare traces precicely

File: frontend/src/pages/Traces/TraceSpanAttributes.tsx [hunk 2]
```diff
@@ -1,14 +1,74 @@
...
+	}
+
```
Comment (Vadman97) [inline frontend/src/pages/Traces/TraceSpanAttributes.tsx:20]: could we also format the `duration` attribute nicely? currently it's just the nanosecond value, but maybe we could label that or convert to seconds and label it `s`?

**Review by Vadman97 — APPROVED** [review]

File: frontend/src/pages/Traces/TraceSpanAttributes.tsx (hunk 2 above)
Comment (ccschmitz) [reply frontend/src/pages/Traces/TraceSpanAttributes.tsx:24]: Handled in https://github.com/highlight/highlight/pull/7517/commits/281149eb92333157202c20762c905e3fc414d780.

File: frontend/src/pages/Traces/TraceSpanAttributes.tsx (hunk 1 above)
Comment (ccschmitz) [reply frontend/src/pages/Traces/TraceSpanAttributes.tsx]: Handled in https://github.com/highlight/highlight/pull/7517/commits/281149eb92333157202c20762c905e3fc414d780.

**Pull request merged**